test/e2e/util.go

```diff
@@ -182,3 +186,121 @@ func randomSuffix() string {
 func expectNoError(err error, explain ...interface{}) {
 	ExpectWithOffset(1, err).NotTo(HaveOccurred(), explain...)
 }
+
+func cleanup(filePath string, selectors ...string) {
+	By("using stop to clean up resources")
+	runKubectl("stop", "-f", filePath)
+
+	for _, selector := range selectors {
+		resources := runKubectl("get", "pods,rc,se", "-l", selector, "--no-headers")
+		if resources != "" {
+			Failf("Resources left running after stop:\n%s", resources)
+		}
+	}
+}
+
+type validatorFn func(c *client.Client, podId string) (interface{}, error)
```

Comment:
I don't see why we'd need to pass the client back to the validatorFn. This can probably be simplified to `type validatorFn func(podID string) error`.